### PR TITLE
Add plotting method for COWC dataset

### PR DESCRIPTION
**Pull request by nilsleh**
Adding plotting method for COWC dataset.

COWCCounting:

cowcCounting2

COWCDetection
cowcDetection2

An Example of image with color:

colorCOWC

### CODE REVIEW

**Comment by adamjstewart** [comment]
* For counting, can you give an example image with # cars > 0? 
* For detection, it might make more sense for the labels (0/1) to be displayed as Boolean (True/False).

File: torchgeo/datasets/cowc.py [hunk 1]
```diff
@@ -250,10 +289,51 @@ class COWCDetection(COWC):
...
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+
+        .. versionadded:: 0.2
+        """
+        image, label = sample["image"], sample["label"]
+
+        label_title = "True" if label == 1 else "False"
```
Comment (adamjstewart) [inline torchgeo/datasets/cowc.py:316]:
```suggestion
        label_title = bool(label)
```

**Review by adamjstewart — COMMENTED** [review]
You might be able to put most of the plotting code in the `COWC` base class, then the subclasses only need to decide what the label/prediction is. If it makes it easier, we could even consider going back from True/False to 1/0 and then we don't need separate plotting code.

**Comment by adamjstewart** [comment]
This looks good to me, but I'll let others review. I'm starting to flip flop, and I think it might be simpler (less code to maintain) if we go back to your first commit where there's a single plot function for the `COWC` dataset. Having detection print 1/0 instead of True/False doesn't seem that important to me when compared to the code duplication.

File: torchgeo/datasets/cowc.py [hunk 2]
```diff
@@ -225,6 +265,40 @@ class COWCCounting(COWC):
...
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+
+        .. versionadded:: 0.2
+        """
+        label_title = cast(str, sample["label"].item())
+
+        showing_predictions = "prediction" in sample
```
Comment (isaaccorley) [inline torchgeo/datasets/cowc.py:288]: Do we need to make this a variable if it's only being used once on the line below?

File: torchgeo/datasets/cowc.py [hunk 3]
```diff
@@ -254,6 +328,40 @@ class COWCDetection(COWC):
...
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+
+        .. versionadded:: 0.2
+        """
+        label_title = cast(str, bool(sample["label"].item()))
+
+        showing_predictions = "prediction" in sample
```
Comment (isaaccorley) [inline torchgeo/datasets/cowc.py:351]: Same here

**Review by isaaccorley — CHANGES_REQUESTED** [review]
I think this lgtm, although I'm curious why the example plots look to be grayscale? Are the images not RGB?

**Comment by adamjstewart** [comment]
> I'm curious why the example plots look to be grayscale? Are the images not RGB?

The images should be RGB, this seems like a bug in the plotting code in this PR.

**Comment by nilsleh** [comment]
I believe that while all images in the dataset have RGB channels, they are not all colored, see the [example](https://gdo152.llnl.gov/cowc/) on the dataset website.

**Comment by adamjstewart** [comment]
Can you show one of the colored images from your plotting method?

File: torchgeo/datasets/cowc.py (hunk 2 above)
Comment (calebrob6) [reply torchgeo/datasets/cowc.py:280]: No, but do we need to change it?

**Comment by adamjstewart** [comment]
I think we should go back to having a single `plot` method in the superclass instead of adding a lot of extra code just to convert 0/1 to False/True.

File: torchgeo/datasets/cowc.py (hunk 2 above)
Comment (isaaccorley) [reply torchgeo/datasets/cowc.py:280]: Nope, I'm indifferent but it should at least be pointed out that it's not consistent with other plot methods.

Comment (calebrob6) [reply torchgeo/datasets/cowc.py:280]: Other plots use this idea https://github.com/microsoft/torchgeo/blob/cbebc1e0dbb77fe37ce4bc8116792828fab8bc27/torchgeo/datasets/eurosat.py#L211

**Review by adamjstewart — APPROVED** [review]

**Review by isaaccorley — APPROVED** [review]

**Pull request merged**